### PR TITLE
scikit-learn-intelex 2023.1.1

### PR DESCRIPTION
**Pull request by skupr-anaconda**
License: https://github.com/intel/scikit-learn-intelex/blob/2023.1.1/LICENSE
Requirements: 
- https://intel.github.io/scikit-learn-intelex/system-requirements.html
- https://github.com/intel/scikit-learn-intelex/blob/2023.1.1/setup_sklearnex.py
- https://github.com/intel/scikit-learn-intelex/blob/2023.1.1/INSTALL.md#build-from-sources

Actions:
1. Enable building the package on `osx-64`
2. Add missing variable `DALROOT" `to `script`, see https://github.com/intel/scikit-learn-intelex/blob/2023.1.1/.ci/pipeline/ci.yml
3. Add `dal-include {{ version }}` to host because the package requires **library_version_info.h** that should be in the `include/services` folder, see https://github.com/intel/scikit-learn-intelex/blob/987b5102fba322d604f29a17db039301e6f8dfbc/scripts/version.py#L39. `dal-include 2023.1.1` contains it
4. Add missing pinning for `scikit-learn` in `run`

### CODE REVIEW

**Comment by skupr-anaconda** [comment]
A variable is missing:
```
Traceback (most recent call last):

  File "/croot/scikit-learn-intelex_1682087022927/work/setup_sklearnex.py", line 45, in <module>

    raise RuntimeError("Not set DALROOT variable")
RuntimeError: Not set DALROOT variable
```

But if I add that variable:
```
if [ -z "${DALROOT}" ]; then
    export DALROOT=${PREFIX}
fi
```
then I got another error:
```
  File "/croot/scikit-learn-intelex_1682089948669/work/setup_sklearnex.py", line 60, in <module>

    ONEDAL_VERSION = get_onedal_version(dal_root)

  File "/croot/scikit-learn-intelex_1682089948669/work/scripts/version.py", line 42, in get_onedal_version

    with open(header_version, 'r') as header:

FileNotFoundError: [Errno 2] No such file or directory: '/croot/scikit-learn-intelex_1682089948669/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placeh/include/services/library_version_info.h'
```

The header `library_version_info.h` should be in the `services` folder
https://github.com/intel/scikit-learn-intelex/blob/987b5102fba322d604f29a17db039301e6f8dfbc/scripts/version.py#L39 but it's not found. Am I doing something wrong or is there a bug in the source code of v2023.1.1?

**Comment by skupr-anaconda** [comment]
The dependant package `scikit-learn-intelex` require **library_version_info.h** and should be in the `services` folder
https://github.com/intel/scikit-learn-intelex/blob/987b5102fba322d604f29a17db039301e6f8dfbc/scripts/version.py#L39 but it's not found. It can be a bug but it's not clear

**Comment by skupr-anaconda** [comment]
I've added `dal-include {{ dal_version }}` to to `host` because `scikit-learn-intelex` requires **library_version_info.h** that should be in the `include/services` folder, 
    # see https://github.com/intel/scikit-learn-intelex/blob/987b5102fba322d604f29a17db039301e6f8dfbc/scripts/version.py#L39.
`dal` headers are inside `dal-include` package https://anaconda.org/anaconda/dal-include/files

**Comment by skupr-anaconda** [comment]
`scikit-learn-intelex` depends on **daal4py 2023.1.1** https://github.com/AnacondaRecipes/daal4py-feedstock/pull/10

File: abs.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+upload_channels:
```
Comment (M-Waszkiewicz-Anaconda) [inline abs.yaml]: why not upload to default ?

Comment (skupr-anaconda) [reply abs.yaml]: It depends on `daal4py` https://github.com/AnacondaRecipes/daal4py-feedstock/pull/10 and -> `mpich` https://github.com/AnacondaRecipes/mpich-feedstock/pull/6

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Comment by skupr-anaconda** [comment]
Fixed the env variable `DALROOT` as:
```
  - export DALROOT="${PREFIX}"  # [not win]
  - set DALROOT=%PREFIX%        # [win]
```

**Comment by skupr-anaconda** [comment]
I'll remove `abs.yaml` before merging the PR

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**